cmd/container-suseconnect/main.go

```diff
@@ -165,11 +173,11 @@ func runZypperURLResolver(_ *cli.Context) error {
 // all available repositories for the installed product. Additional modules
 // can be specified via the `ADDITIONAL_MODULES` environment variable, which
 // reflect the module `identifier`.
-func runZypperPlugin(_ *cli.Context) error {
+func runZypperPlugin(ctx *cli.Context) error {
 	log.SetOutput(cs.GetLoggerFile())
 
 	products, err := requestProducts()
-	if err != nil {
+	if ctx.Bool("fail-on-error") && err != nil {
```

Comment:
Shouldn't this be done in the calling context?

Reply:
I don't understand what you mean by that

Reply:
previously, when an error was set, it was exiting the function here and not executing the remaining part of the code. now with your change it continues execution of the function, which is a behavior change, and I fear an unintended change.

if the purpose is to simply suppress the error handling, then the error should be returned here like before, and in the calling context that prints/exits the error it should do $therightthing to suppress it. 

this would also have the advantage that the `--fail-on-error` option also works for actions other than the one you just patched. with your change it only works in one out of 3 cases that exist in the code. 

Following DRY principle, I'm leaning towards not handling the flag in 3 cases.